### PR TITLE
Custom form 774

### PR DESCRIPTION
**Pull request by popkinj**
-  Custom form defined by a JSON config file.
-  Docker volume mount to replace the forms directory in the public Vue space.
-  Helm config to mount the forms JSON **I'm half expecting this to explode** Planning to test in the deployed PR.

### CODE REVIEW

File: scripts/docker-compose.yml
Comment (loneil) [inline scripts/docker-compose.yml:220]: Probably merge problem here. This needs to stay as 0.10.3

File: charts/traction/templates/ui/deployment.yaml
Comment (i5okie) [inline charts/traction/templates/ui/deployment.yaml:76]: `name` here is the name of the volume defined within `volumes` on line 82.
Setting it to the name of the config map will probably work, as it will resolve to a fully qualified object name.
It is not necessary however, I would name it something descriptive, for example `reservation-form`.

File: charts/traction/templates/ui/deployment.yaml
Comment (i5okie) [inline charts/traction/templates/ui/deployment.yaml:82]: same as above

File: charts/traction/values.yaml
Comment (i5okie) [inline charts/traction/values.yaml:1]: `reservationForm` should be defined and documented in values.yaml, even as an empty object.

**Review by i5okie — CHANGES_REQUESTED** [review]

File: services/tenant-ui/config/custom-environment-variables.json
Comment (loneil) [inline services/tenant-ui/config/custom-environment-variables.json:8]: This isn't used anywhere?

File: charts/traction/templates/ui/configmap.yaml
Comment (loneil) [inline charts/traction/templates/ui/configmap.yaml:17]: This isn't used anywhere?

File: scripts/docker-compose.yml
Comment (loneil) [inline scripts/docker-compose.yml:310]: Should stay .3

File: deploy/traction/values-pr.yaml
Comment (loneil) [inline deploy/traction/values-pr.yaml:62]: Is this to be JSON in the YAML file or should it be YAML? Wondering Ivan's opinion.

File: scripts/docker-compose.yml
Comment (popkinj) [reply scripts/docker-compose.yml:310]: Sigh... Why does this keep reverting itself.

File: deploy/traction/values-pr.yaml
Comment (popkinj) [reply deploy/traction/values-pr.yaml:62]: Yes... this is supposed to be JSON

File: scripts/docker-compose.yml
Comment (popkinj) [reply scripts/docker-compose.yml]: Done.

File: deploy/traction/values-pr.yaml
Comment (loneil) [reply deploy/traction/values-pr.yaml]: 👍

File: services/tenant-ui/config/custom-environment-variables.json
Comment (popkinj) [reply services/tenant-ui/config/custom-environment-variables.json:8]: Yup. In the configmap.yaml.

File: charts/traction/templates/ui/configmap.yaml
Comment (popkinj) [reply charts/traction/templates/ui/configmap.yaml:17]: Same as above.

**Comment by popkinj** [comment]
The Helm integration for the custom form is finally working.
Screenshot from 2023-10-12 11-46-15

File: charts/traction/templates/ui/configmap.yaml
Comment (loneil) [reply charts/traction/templates/ui/configmap.yaml:17]: But where is this env var used? I just pulled the code and I don't see it anywhere? Aren't you using a file instead of an environment variable?

File: services/tenant-ui/config/custom-environment-variables.json
Comment (loneil) [reply services/tenant-ui/config/custom-environment-variables.json:8]: See comment there. I don't actually see `customReservationForm` used anywhere in the app. It's a file, not an env var isn't it?

Comment (popkinj) [reply services/tenant-ui/config/custom-environment-variables.json:8]: Hmmmm.... 
It's actually a combination of a file and Helm config.
I'll have a look. :eyes:

Comment (popkinj) [reply services/tenant-ui/config/custom-environment-variables.json]: Nice catch @loneil .

**Comment by loneil** [comment]
Playing with another value in values-pr, I will put back to as it was before before we merge

**Review by loneil — APPROVED** [review]
Think this is looking good to me. Tested some cases on local Docker and on the PR and working for me.

I'd defer to @i5okie for a final approval with the Helm config set up though.

I created a ticket here to track what we should set for the other env vals
https://github.com/bcgov/traction/issues/868
Might be worth handling the DEV one sooner than later (could be in this PR if we know what it should be) as the default would take effect on merge of this.

**Pull request merged**
